ci-operator/step-registry/platformoperator/valid/cert-manager/platformoperator-valid-cert-manager-commands.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+set -euo pipefail
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+run_command "oc version -o yaml"
+
+semver_compare() {
+  local version_a version_b pr_a pr_b
+  # strip word "v" and extract first subset version (x.y.z from x.y.z-foo.n)
+  version_a=$(echo "${1//v/}" | awk -F'-' '{print $1}')
+  version_b=$(echo "${2//v/}" | awk -F'-' '{print $1}')
+
+  if [ "$version_a" \= "$version_b" ]
+  then
+    # check for pre-release
+    # extract pre-release (-foo.n from x.y.z-foo.n)
+    pr_a=$(echo "$1" | awk -F'-' '{print $2}')
+    pr_b=$(echo "$2" | awk -F'-' '{print $2}')
+
+    ####
+    # Return 0 when A is equal to B
+    [ "$pr_a" \= "$pr_b" ] && echo 0 && return 0
+
+    ####
+    # Return 1
+
+    # Case when A is not pre-release
+    if [ -z "$pr_a" ]
+    then
+      echo 1 && return 0
+    fi
+
+    ####
+    # Case when pre-release A exists and is greater than B's pre-release
+
+    # extract numbers -rc.x --> x
+    number_a=$(echo ${pr_a//[!0-9]/})
+    number_b=$(echo ${pr_b//[!0-9]/})
+    [ -z "${number_a}" ] && number_a=0
+    [ -z "${number_b}" ] && number_b=0
+
+    [ "$pr_a" \> "$pr_b" ] && [ -n "$pr_b" ] && [ "$number_a" -gt "$number_b" ] && echo 1 && return 0
+
+    ####
+    # Retrun -1 when A is lower than B
+    echo -1 && return 0
+  fi
+  arr_version_a=(${version_a//./ })
+  arr_version_b=(${version_b//./ })
+  cursor=0
+  # Iterate arrays from left to right and find the first difference
+  while [ "$([ "${arr_version_a[$cursor]}" -eq "${arr_version_b[$cursor]}" ] && [ $cursor -lt ${#arr_version_a[@]} ] && echo true)" == true ]
+  do
+    cursor=$((cursor+1))
+  done
+  [ "${arr_version_a[$cursor]}" -gt "${arr_version_b[$cursor]}" ] && echo 1 || echo -1
+}
+
+###
+# K8s v1.23 = OCP v4.10
+# K8s v1.24 = OCP v4.11
+# K8s v1.25 = OCP v4.12
+# ...
+###
+# Get K8s version from the oc client
+k8s_version=$(oc version | grep Kubernetes |awk '{print $3}')
```

Comment:
@jianzhangbjz it seems you depend on oc binary to check version of the target cluster, so,
1, better to inject oc to step.
current defintion
```yaml
ref:
  as: platformoperator-valid-cert-manager
  commands: platformoperator-valid-cert-manager-commands.sh
  from: cli
```
does take correct oc although [here](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A14) show the cli image for 4.14.
[the log ](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A48-68)show its client is 4.2.

2, there is [error log](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A47). 

3, per  [the log ](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A48-68), there is no keyword `Kubernetes`. it is keyword `gitVersion` for both client and server.

Reply:
> [the log ](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A48-68)show its client is 4.2.

Yes, the client always shows 4.2, but I would like to use the `gitVersion` of `serverVersion`.
```yaml
clientVersion:
  buildDate: "2023-03-09T14:29:35Z"
  compiler: gc
  gitCommit: 07ee29c0d
  gitTreeState: dirty
  gitVersion: v4.2.0-alpha.0-17[50](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A50)-g07ee29c
  goVersion: go1.19.4
  major: ""
  minor: ""
  platform: linux/amd64
kustomizeVersion: v4.5.7
serverVersion:
  buildDate: "2023-02-21T11:03:12Z"
  compiler: gc
  gitCommit: b6d1f054747e9886f61dd8[53](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37211/rehearse-37211-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-proxy-techpreview-p3-f28-destructive/1635490201935548416#1:build-log.txt%3A53)16deac3415e2726f
  gitTreeState: clean
  gitVersion: v1.25.4+18eadca
  goVersion: go1.19.4
  major: "1"
  minor: "25"
  platform: linux/amd64
```